persistent/_timestamp.c

```diff
@@ -152,7 +152,12 @@ TimeStamp_richcompare(TimeStamp *self, TimeStamp *other, int op)
     return result;
 }
 
+
+#ifdef PY3K
+static Py_hash_t
+#else
 static long
+#endif
```

Comment:
Nice catch!  Logically, this kind of switch would be better as a `#define` in https://github.com/zopefoundation/persistent/blob/master/persistent/_compat.h.